src/Control/Lens/Traversal.hs

```diff
@@ -1466,3 +1478,82 @@ traverseByOf l pur app f = reifyApplicative pur app (l (ReflectedApplicative #.
 -- @
 sequenceByOf :: Traversal s t (f b) b -> (forall x. x -> f x) -> (forall x y. f (x -> y) -> f x -> f y) -> s -> f t
 sequenceByOf l pur app = reifyApplicative pur app (l ReflectedApplicative)
+
+-- | A version of 'Control.Lens.Setter.over' that forces the result(s) of
+-- applying the function. This can prevent space leaks when modifying lazy
+-- fields. See also 'strictly'.
+--
+-- @
+-- over traverse (const ⊥) [1,2] = [⊥, ⊥]
+-- over' traverse (const ⊥) [1,2] = ⊥
+-- @
+--
+-- @
+-- over' :: 'Lens' s t a b -> (a -> b) -> s -> t
+-- over' :: 'Traversal' s t a b -> (a -> b) -> s -> t
+-- @
+--
+-- >>> over traverse (\x -> if x > 1 then undefined else pure x) [1,2]
+-- [Exc
+-- >>> over' traverse (\x -> if x > 1 then undefined else pure x) [1,2]
+-- Exc
+over' :: LensLike Solo s t a b -> (a -> b) -> s -> t
+over' l f = getSolo . l (\old -> Solo $! f old)
+{-# INLINE over' #-}
+
+-- | A version of 'Control.Lens.Setter.iover' that forces the result(s) of
+-- applying the function. Alternatively, an indexed version of `over'`.
+-- See also 'strictly'.
+--
+-- >>> iover' :: IndexedLens s t a b -> (i -> a -> b) -> s -> t
+-- iover' ::
+-- >>> iover' :: IndexedTraversal s t a b -> (i -> a -> b) -> s -> t
+-- iover' ::
+-- @
+iover' :: Over (Indexed i) Solo s t a b -> (i -> a -> b) -> s -> t
+iover' l f = getSolo . l (Indexed $ \i a -> Solo $! f i a)
+{-# INLINE iover' #-}
+
+-- | Use an optic /strictly/. This will force the results of /all/ the targets
+-- before producing a new value. Note that producing an optic using 'strictly'
+-- will not necessarily produce one even nearly as efficient as what could be
+-- written by hand, although it will do so in simple enough situations. Be
+-- particularly careful when working over many targets in a functor other than
+-- 'Identity' or @'Const' c@; things may go very well or very badly.
+--
+-- @
+-- 'over'' l = 'Control.Lens.Setter.over' (strictly l)
+-- 'iover'' l = 'Control.Lens.Setter.iover' (strictly l)
+-- @
+--
+-- @
+-- strictly :: 'Traversal' s t a b -> 'Traversal' s t a b
+-- strictly :: 'IndexedTraversal' i s t a b -> 'IndexedTraversal' i s t a b
+-- @
+strictly :: (Functor m, Profunctor p, Profunctor q) => Optical p q (BoxT m) s t a b -> Optical p q m s t a b
+strictly l f = rmap (fmap getSolo .# runBoxT) $ l (rmap (\mb -> BoxT ((Solo $!) <$> mb)) f)
+{-# INLINE strictly #-}
+
+-- | A very simple applicative transformer that gives us more control over when
+-- things get forced. Note: this type /should not/ be made an instance of
+-- @Settable@, because then users could accidentally use 'strictly' with a
+-- 'Setter', which will not work at all. There is no way to strictify a
+-- 'Setter'.
+newtype BoxT m a = BoxT
+  { runBoxT :: m (Solo a) }
+  deriving (Functor, Foldable, Traversable)
```

Comment:
Where should this type be defined and exported?

Reply:
So you don't want to wait until `SoloT` is added somewhere, as you proposed in private mail to me and Ross (maintainer of `transformers`)?

I understand that you have time to work on stuff during holidays, but I honestly don't, so will politely ignore you until the next year.

Reply:
`SoloT` turned out to be the wrong sort of transformer for this purpose. I made a mistake and changed course.

Reply:
Please tell that to others in that email message too. I wasn't aware that I don't need to think about that anymore.

Reply:
I learned yesterday that this idea has been invented before (rather unsurprisingly); I've edited, expanded, and adapted a note about it in `optics` for inclusion here.

Reply:
Done.